### PR TITLE
[MARCHETYPES-76] Update project to Java 8

### PR DESCRIPTION
**Pull request by slachiewicz**
- test also with Java 21

### CODE REVIEW

File: .github/workflows/maven-verify.yml [hunk 1]
```diff
@@ -25,3 +25,5 @@ jobs:
   build:
     name: Verify
     uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v3
+    with:
+      jdk-matrix: '[ "8", "11", "17", "21" ]'
```
Comment (judovana) [inline .github/workflows/maven-verify.yml:29]: This shoudl be standalone change

Comment (slachiewicz) [reply .github/workflows/maven-verify.yml:29]: this change is required to show that with Java 21 we have issue to run integration tests

File: maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
-#macro( compilerProperties $jcv )
+#macro( compilerProperties $javaCompilerVersion )
```
Comment (judovana) [inline maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml:2]: please do not revert it. As explained and agreed in: https://github.com/apache/maven-archetypes/pull/18#discussion_r1432919700

File: .github/workflows/maven-verify.yml (hunk 1 above)
Comment (judovana) [reply .github/workflows/maven-verify.yml:29]: Sure it is. And I agree wiht it. But not in the way you did it.
You had mixed three parts in one PR I do not understand why. This should be separate PR, whcihshoudl go in before https://github.com/apache/maven-archetypes/pull/20

Aslo you are reverting the change which was agreed during review of #18

**Comment by judovana** [comment]
Why are you reverting the test change? The testchange was correct. Now

File: maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml (hunk 2 above)
Comment (slachiewicz) [reply maven-archetype-quickstart/src/main/resources-filtered/archetype-resources/pom.xml:2]: commit reverted - please open new with all correct changes

**Pull request closed without merging**